pkg/daemon/metrics.go

```diff
@@ -0,0 +1,189 @@
+package daemon
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+	"net/http"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	PTPNamespace = "openshift"
+	PTPSubsystem = "ptp"
+
+	ptp4lProcessName = "ptp4l"
+	phc2sysProcessName = "phc2sys"
+)
+
+var (
+	NodeName = ""
+
+	OffsetFromMaster = prometheus.NewGaugeVec(
```

Comment:
When ptp processes are terminated, will these metrics become stale?
Do you think we shall update these metrics to default non-zero values?